lib/stdlib/test/ets_SUITE.erl

```diff
@@ -3068,14 +3070,24 @@ write_concurrency(Config) when is_list(Config) ->
 
     true = YesMem > YesTreeMem,
 
-    case erlang:system_info(schedulers) > 1 of
-        true ->
+    case erlang:system_info(schedulers) of
+        1 ->
+            YesMem = NoHashMem,
+            YesTreeMem = NoTreeMem,
+            YesYesTreeMem = YesTreeMem;
+        NoSchedulers ->
             true = YesMem > NoHashMem,
             true = YesMem > NoTreeMem,
-            true = YesTreeMem < NoTreeMem,
-            true = YesYesTreeMem > YesTreeMem;
-        _ ->
-            one_scheduler_only
+
+            %% The memory of ordered_set with write concurrency is
+            %% smaller than without write concurrency on systems with
+            %% few schedulers.
```

Comment:
The non-ca implementation pre-allocates memory to use as the traversal stack, so that uses a bit of extra memory for non-ca. As the number of schedulers grow, the base node rw lock in the ca implementation starts using more and more memory, and at 6 schedulers that memory is the same as the size of the pre-allocated traversal stack.